cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/DefaultPluginConfigurer.java

```diff
@@ -82,6 +84,7 @@ public final <T> T usePlugin(String pluginType, String pluginName, String plugin
       Plugin plugin = addPlugin(pluginType, pluginName, pluginId, properties, selector);
       return pluginInstantiator.newInstance(plugin);
     } catch (PluginNotExistsException | IOException e) {
+      LOG.error("Plugin name: {} with plugin type: {} cannot be added or instantiated", pluginName, pluginType, e);
```

Comment:
it's not good to log an error for PluginNotExistsException, as it may not be an error to whoever is calling the method.

IOException seems like it should always be unexpected though, so let's catch the IOException separately and log that one before returning null.

Similar behavior should be done for the other usePluginClass() method

Reply:
should I throw it or return null?

Reply:
just return null for now, as throwing it would change the behavior